test/ReactSetStateTestComponent2.js

```diff
@@ -79,6 +79,14 @@ class ReactSetStateTestComponent2 extends React.Component {
         this.setState(this.innerTransactionalSetStateCallback.bind(this), this.innerSetStateCallback.bind(this));
     }
 
+    componentDidCatch(_, __) {
+        this.recordLifecycleCall('componentDidCatch');
+    }
+
+    getDerivedStateFromError(_) {
+        this.recordLifecycleCall('getDerivedStateFromError');
```

Comment:
Should we return a map from this to test / assert that the state is updated with the returned value?

Reply:
I definitely should have done that. To fix it, I ended up mirroring the `component2.dart` functionality that relies on `getDerivedStateFromError` returning a state update correctly in order to render properly. I feel like this and the error boundary test in the `Component2` group of `lifecycle_test.dart` show that the state is updating correctly, but let me know if I misunderstood or there's a different way to approach testing that would add more coverage!